### DISCUSSION TITLE
idea : consistency in hotkeys for opening editing dialogs

### DISCUSSION THREAD

**Discussion by kenfhill84083** [freeplane/freeplane#1319] [General]
I have remapped the hotkeys for the editor dialogs (details & notes) to match the default node core dialog (Alt+Enter).  Details is Alt+Plus and notes is Alt+Minus.  see image

Over the years I've tried different key mappings for the dialogs but this is - for me - more logical and, therefore, less confusing.  I've tried it for a few weeks now and it just feels right.  Maybe this will resonate with someone else in Freeplane Land.

Of course, mileage may vary.

![freeplane _ hotkeys for consistency opening editing dialogs](https://github.com/freeplane/freeplane/assets/6380301/c50d19e6-26c1-4b1c-afd7-1f810c8597f4)

**Comment by sifran-github** [comment]
Personally, I almost never use dialogs. 98% of the time I use the regular editing for NOTES (F2) and DETAILS (F3), and I always display the notes panel on the right side so I edit them there (View->Notes->Note panel Position->right). So I rarely use any dialog, and when i do, i just right click a node and choose.

I understand the usage of dialogs when the nodes/details are complicated, but personally I try to keep the notes and details very minimalistic, with just a few words, and have additional information in notes.

but one of the great features of Freeplane is: its very customizable, so different users can use it for their specific usage :)

**Reply by kenfhill84083** [reply]
I, too, rejoice in Freeplane's robust flexibilty that makes it functional and efficient for all kinds of users.  It is cool that users can develop their own small tweaks to make their work easier.

Like you, I often keep the notes panel open (usually on the left).   Sometimes I park on the current note by turning off "View->Notes->Note follows selection" - so I have a persistent reference for breaking apart its text into a map branch of keyword nodes.  Sometimes, though, I want a node in the developing branch to have its own note.  That is one instance where the notes dialog comes in handy.

I use FP for Bible study notes.  To minimize distraction, I like verse numbers to be smaller than the Bible text.  I find using dialogs to be handy for that because I hide the toolbar in the notes panel (increasing notes real estate).  It is easier for me to change font size with the toolbar - always visible in the dialog - than with the menu: Format > Font...

**Comment by euu2021** [comment]
What I do is concentrating Inline and Dialog editing around the same F key. So, Inline core is F2, and Dialog core is Shift+F2. This way, I memorize two hotkeys for the memory price of one 🤣

Also, I assign the sequence of the F keys to Core/Details/Note/Attributes. For some reason, this sequence feels intuitive to me.

So, the result is this:
![image](https://github.com/freeplane/freeplane/assets/77707706/ee51db5a-e85b-41d2-b6b1-d403c49c797c)

But, the reality is that, 99% of the time, I only use Inline Core and Inline Attributes. So, I should probably leave the F keys for something more useful to me 🤔

> [...] Freeplane Land [...]

🤣🤣🤣

**Reply by kenfhill84083** [reply]
Thanks for the reply.  I like the chart.  I may play around with such a table in RightNote (my go-to table editing app).  I have something like your remapping for f3, including alt+f3 for displaying/hiding details.

I don't use attributes much but I should investigate it.  I've watched some of Quinbus' videos and been impressed by how he uses attributes.  I haven't seen yet how it would work for my less complex uses.

I built my dialogs system around alt+Enter because (1) it was FP's default and (2) I already use it in Windows file-managing apps (eg Explorer, Everything).

To access core text inline I generally use Home or End to quickly get to the beginning or the end respectively.  (I have been using it so long I can't remember if that is default or something I remapped.)

It is interesting to see how others users' minds work for something like this.

**Reply by kenfhill84083** [reply]
You mentioned you could get rid of Shift+Fkey hotkeys since you rarely use them.  I have run into this also.  Now, with Command Search, I can access many less-used functions quickly and painlessly without having to memorize hotkeys for seldom-used ones.  I love Command Search.

**Reply by euu2021** [reply]
> [...] my go-to table editing app [...]

I create a lot of tables. I like to create on LibreOffice Writer and store on FP (by pasting the HTML). Then I can do simple edits directly in FP, or, for more complex edits, I paste back in LibreOffice. And the contents of the table show up when I do a global search in my map (I use a single big map for all my contents).

 I just checked the tables in RightNote here, and it works well, too. I didn't like how it works with footnotes, though. I didn't find a way to show all the footnotes at the same time. I like to use footnes, in order to keep the table clean. For example (notice that FP can store footnotes well, too):

![image](https://github.com/freeplane/freeplane/assets/77707706/7bc989c6-e318-405b-b0fc-a3ce7e1ea9e2)

Anyway, I will leave the LibreOffice Writer as a suggestion of a free alternative. I tested many programs, and it seems to be the best to create tables.

> You mentioned you could get rid of Shift+Fkey hotkeys since you rarely use them. I have run into this also. Now, with Command Search, I can access many less-used functions quickly and painlessly without having to memorize hotkeys for seldom-used ones. I love Command Search.

Yes, I use Command Search a lot, to avoid having to memorize the hotkeys. And I use many undocked views, so it gets even more essential.

**Reply by kenfhill84083** [reply]
Freeplane offers a lot of levels of meaning you can pack into or associate with a node.

But your tables are a whole new level for me.  WOW!  I use LO Writer a lot, too, but I never dreamed about such integration with Freeplane.  For my much simpler tables, and RightNote's highly remappable hotkeys, I am pretty agile.  I'll have to see if I can streamline Writer in a similar way.  Do you have any table hotkeys?

Thanks a lot for this useful post.

**Reply by euu2021** [reply]
I don't use hotkeys for tables. I use only the contextual toolbar that shows up when a table is active. This toolbar can be customizable to have only the icons that are actually useful in daily life, so the operation gets efficient even without hotkeys.

Customizing the style of the content of the table is helpful, too, so you don't need to format it every time.
